### ISSUE TITLE
onUserMediaError parameter type is wrong in .d.ts file

### ISSUE THREAD

**Issue by mwarnerdotme** [mozmorris/react-webcam#272]
## Please follow the general troubleshooting steps first:

- [ yes ] Is your app running over HTTPS? (please provide the URL if possible)
- [ yes ] Have you tried running the demo (https://codepen.io/mozmorris/pen/JLZdoP) on your device?
- [ yes ] Checked the latest "Can I use" compatbility table? (https://caniuse.com/stream)

### Bug reports:

The functionality is as expected, but I am forced to cast the `string` to a `DOMException` to avoid any errors in my compilation and get the functionality that I need.

#### Example
At first, I was calling this code:
```javascript
function onUserMediaError(error: string) {
  if (error.includes('device not found')) {
    history.push('/login/no-camera')
  }
}
```

However, this code would result in an error stating that `error` did not have an `includes` property or function. It doesn't because it's a `DOMException` even though I have typed my parameter (`error`) to a `string` type. This compiles properly, but the types do not match up.

#### Expected
After realizing what is actually returned (a `DOMException`), the type in the `.d.ts` file should be changed to accept a function with a parameter of type `DOMException` instead of type `string`. After changing the `.d.ts` provided in version `5.2.2`, I was able to run this code perfectly fine:
```javascript
function onUserMediaError(error: DOMException) {
  if (error.message.includes('device not found')) {
    history.push('/login/no-camera')
  }
}
```

**Comment by mozmorris** [comment]
Good catch @mwarnerdotme. I think you'll need to make it a union type of `string | DOMException`

**Comment by mozmorris** [comment]
released in `5.2.3`
